src/test/java/com/sap/sgs/phosphor/fosstars/data/github/TestGitHubDataFetcherHolder.java

```diff
@@ -53,42 +48,89 @@ public void init() throws IOException {
   @After
   public void cleanup() throws IOException {
     List<Path> deletedPaths = new ArrayList<>();
-    fetcher.cleanup(((url, repository, total) -> {
+    fetcher.cleanup((url, repository, total) -> {
       deletedPaths.add(repository.path());
       return true;
-    }));
+    });
 
     for (Path deletedPath : deletedPaths) {
       assertFalse(Files.exists(deletedPath));
     }
 
-    FileUtils.forceDeleteOnExit(path.toFile());
+    FileUtils.forceDeleteOnExit(fetcher.baseDir().toFile());
   }
 
-  public static class TestGitHubDateFetcher extends GitHubDataFetcher {
+  public static class TestGitHubDataFetcher extends GitHubDataFetcher {
 
-    public TestGitHubDateFetcher(GitHub github, Path base) throws IOException {
-      super(github, base);
+    /**
+     * Test class constructor.
+     */
+    public TestGitHubDataFetcher(GitHub github) throws IOException {
+      super(github);
     }
 
+    /**
+     * Adds {@link GitHubProject} and its associated {@link LocalRepository} details to cache.
+     * @param project The {@link GitHubProject}.
+     * @param repository The {@link LocalRepository}.
+     */
     public void addForTesting(GitHubProject project, LocalRepository repository) {
-      localRepositories.put(project, repository);
+      LOCAL_REPOSITORIES.put(project, repository);
+    }
+
+    /**
+     * Returns the current list of local repositories stored in the cache.
+     *
+     * @return Map of local repositories in cache.
+     */
+    public Map<GitHubProject, LocalRepository> getLocalRespositoriesForTesting() {
+      return LOCAL_REPOSITORIES;
+    }
+
+    /**
+     * Returns the current list of local repositories stored in the base folder.
+     *
+     * @return Map of local repositories available.
+     * @throws IOException if something goes wrong.
+     */
+    public Map<URL, LocalRepositoryInfo> getLocalRespositoriesInfoForTesting() throws IOException {
+      return LOCAL_REPOSITORIES_INFO;
+    }
+    
+    /**
+     * Add new project to be considered while loading repository.
+     *  
+     * @param project The {@link GitHubProject}.
+     * @param projectDir The local {@link Path} for the {@link GitHubProject}.
+     */
+    public void addRepositoryInfoForTesting(GitHubProject project, Path projectDir) {
+      LOCAL_REPOSITORIES.remove(project);
+      LOCAL_REPOSITORIES_INFO.put(project.url(),
+          new LocalRepositoryInfo(projectDir, Date.from(Instant.now()), project.url()));
     }
 
     @Override
     protected void clone(GitHubProject project, Path base) {
       // do nothing
     }
 
-    @Override
-    protected Map<URL, LocalRepositoryInfo> loadLocalRepositoriesInfo() {
-      return new HashMap<>();
+    /**
+     * Returns a resolved valid path of the project.
+     *  
+     * @param projectName The name of the project.
+     * @return path of type {@link Path}. 
+     */
+    static Path projectDir(String projectName) {
+      return DEFAULT_DIRECTORY.resolve(projectName);
```

Comment:
How about the following? I think it would make the code a bit shorter.

```suggestion
    static Path directoryFor(GitHubProject project) {
      return DEFAULT_DIRECTORY.resolve(project.name());
```